mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -259,6 +259,46 @@ LogicalResult SYCLIDConstructorOp::verify() {
       "expects a different signature. Check documentation for details");
 }
 
+bool SYCLWrapOp::areCastCompatible(TypeRange inputs, TypeRange outputs) {
+  if (inputs.size() != 1 || outputs.size() != 1)
+    return false;
+
+  Type sourceType = inputs.front();
+  Type resultType = outputs.front();
+  Type bodyType =
+      TypeSwitch<Type, Type>(resultType)
+          .Case<HalfType>([](auto ty) { return ty.getBody().front(); })
```

Comment:
We try to avoid using `getBody` whenever possible. Can we return `f16` here?

Reply:
👍, I introduced a new helper `getBodyType` for that (naming is hard).